src/System.Reflection/tests/EventInfoTests.cs

```diff
@@ -0,0 +1,196 @@
+            Assert.Throws(exceptionType, () => eventInfo.AddEventHandler(target, handler));
+        }
+
+        public static IEnumerable<object[]> RemoveEventHandler_Invalid_TestData()
+        {
+            // Target is null
+            yield return new object[] { typeof(BaseClass), nameof(BaseClass.PublicEvent), new BaseClass(), new EventHandler(ObjectEventArgsHandler), null, new EventHandler(ObjectEventArgsHandler), typeof(TargetException) };
+
+            // Target is incorrect
+            yield return new object[] { typeof(BaseClass), nameof(BaseClass.PublicEvent), new BaseClass(), new EventHandler(ObjectEventArgsHandler), "hello", new EventHandler(ObjectEventArgsHandler), typeof(TargetException) };
+
+            // Handler is incorrect
+            yield return new object[] { typeof(BaseClass), nameof(BaseClass.PublicEvent), new BaseClass(), new EventHandler(ObjectEventArgsHandler), new BaseClass(), new ObjectDelegate(ObjectHandler), typeof(ArgumentException) };
+        }
+
+        [Theory]
+        [MemberData(nameof(RemoveEventHandler_Invalid_TestData))]
+        public void RemovEventHandler_Invalid(Type type, string name, object addTarget, Delegate addHandler, object removeTarget, Delegate removeHandler, Type exceptionType)
+        {
+            EventInfo eventInfo = GetEventInfo(type, name);
+            eventInfo.AddEventHandler(addTarget, addHandler);
+
+            Assert.Throws(exceptionType, () => eventInfo.RemoveEventHandler(removeTarget, removeHandler));
+        }
+
+        [Theory]
+        [InlineData(typeof(BaseClass), nameof(BaseClass.PublicEvent), typeof(BaseClass), nameof(BaseClass.PublicEvent), true)]
+        [InlineData(typeof(BaseClass), nameof(BaseClass.PublicEvent), typeof(SubClass), nameof(SubClass.PublicEvent), false)]
+        [InlineData(typeof(BaseClass), nameof(BaseClass.PublicEvent), typeof(BaseClass), nameof(BaseClass.PublicStaticEvent), false)]
+        public void Equals(Type type1, string name1, Type type2, string name2, bool expected)
+        {
+            EventInfo eventInfo1 = GetEventInfo(type1, name1);
+            EventInfo eventInfo2 = GetEventInfo(type2, name2);
+            Assert.Equal(expected, eventInfo1.Equals(eventInfo2));
+            Assert.Equal(expected, eventInfo1.GetHashCode().Equals(eventInfo2.GetHashCode()));
+        }
+        
+        [Theory]
+        [MemberData(nameof(Events_TestData))]
+        public void EventHandlerType(Type type, string name)
+        {
+            EventInfo eventInfo = GetEventInfo(type, name);
+            Assert.Equal(typeof(EventHandler), eventInfo.EventHandlerType);
+        }
+
+        [Theory]
+        [MemberData(nameof(Events_TestData))]
+        public void Attributes(Type type, string name)
+        {
+            EventInfo eventInfo = GetEventInfo(type, name);
+            Assert.Equal(EventAttributes.None, eventInfo.Attributes);
+            Assert.False(eventInfo.IsSpecialName);
```

Comment:
Well, the code in coreclr is:
```
        public bool IsSpecialName 
        {
            get 
            {
                return(Attributes & EventAttributes.SpecialName) != 0;
            }
        }
```

So I saw an implicit connection between Attributes and IsSpecialName, so they could be tested together - let me know if you'd still like me to change this